.github/workflows/ci-test.yml

```diff
@@ -366,3 +366,27 @@ jobs:
           if [[ "$success" != "true" ]]; then
             exit 1;
           fi
+
+  perf-benchmarks:
+    name: run-perf-benchmarks
+    runs-on: ubuntu-latest-xl
+    timeout-minutes: 30
+    needs:
+      - build-benchmarks
+    env:
+      CRITERION_HOME: criterion
+    steps:
+      - name: Download the previous baseline
```

Comment:
You'd need checkout and pre build steps.  See examples in code block starting line 304 above.